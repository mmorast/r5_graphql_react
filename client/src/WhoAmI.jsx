import React from 'react'
import {graphql, gql} from 'react-apollo'
import './WhoAmI.css'

const WhoAmIQuery = gql`query WhoAmi {viewer {name email}}`

const WhoAmI = props => {
  const {data} = props
  const {loading, error, viewer} = data

  if (loading) return <p>Loading...</p>

  if (error) return <p>{error.message}</p>

  if (viewer === null) return <p>No viewer returned (which is weird)</p>

  return (
    <div className="WhoAmI">
      <h2>Who Am I?</h2>
      <ul className="WhoAmI__List">
        <li><strong>Name:</strong> {viewer.name}</li>
        <li><strong>Email:</strong> {viewer.email}</li>
      </ul>

    </div>
  )
}


const WhoAmIWithData = graphql(WhoAmIQuery)(WhoAmI)

export default WhoAmIWithData
