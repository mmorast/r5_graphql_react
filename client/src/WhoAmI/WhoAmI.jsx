import React from 'react'
import PropTypes from 'prop-types'
import {
  Col,
  Row,
  Grid,
  Panel,
  ListGroup,
  ListGroupItem
} from 'react-bootstrap'

const WhoAmi = props => {
  const {viewer: {self: viewer}} = props
  return(
    <Grid fluid={true}>
      <Row>
        <Col sm={12}>
          <Panel header={'Who am I?'}>
            <ListGroup>
              <ListGroupItem>Name: {viewer.name}</ListGroupItem>
              <ListGroupItem>Email: {viewer.email}</ListGroupItem>
            </ListGroup>
          </Panel>
        </Col>
      </Row>

    </Grid>
  )
}

WhoAmi.propTypes = {
  viewer: PropTypes.object.isRequired,
}

export default WhoAmi
