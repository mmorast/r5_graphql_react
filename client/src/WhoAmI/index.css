.WhoAmI {
  margin: auto;
  width: 70%;
  text-align: left;
}

.WhoAmi__List {

}
