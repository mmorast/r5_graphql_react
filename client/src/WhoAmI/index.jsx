import React from 'react'
import {graphql} from 'react-apollo'
import gql from 'graphql-tag'
import './index.css'
import WhoAmI from './WhoAmI'

const WhoAmIQuery = gql`query WhoAmi {viewer {name email}}`

export const WhoAmIContainer = props => {
  const {data} = props
  const {loading, error, viewer} = data

  if (loading) return <p>Loading...</p>

  if (error) return <p>{error.message}</p>

  if (viewer === null) return <p>No viewer returned (which is weird)</p>

  return (<WhoAmI viewer={viewer}/>)
}

export default graphql(WhoAmIQuery)(WhoAmIContainer)
