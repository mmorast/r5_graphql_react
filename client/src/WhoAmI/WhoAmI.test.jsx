import React from 'react'
import {shallow} from 'enzyme'
import WhoAmI from './WhoAmI'

test("Renders the WhoAmI component", ()=>{
  let actual = shallow(<WhoAmI viewer={ {self: {name: 'Fred', email: 'fred_murtz@lucy.com'} } }/>)
  expect(actual).toMatchSnapshot()
})
