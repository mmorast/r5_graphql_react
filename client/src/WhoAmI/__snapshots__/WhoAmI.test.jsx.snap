// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the WhoAmI component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WhoAmi
    viewer={
        Object {
            "email": "fred_murtz@lucy.com",
            "name": "Fred",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": <Row
        bsClass="row"
        componentClass="div"
>
        <Col
                bsClass="col"
                componentClass="div"
                sm={12}
        >
                <Panel
                        bsClass="panel"
                        bsStyle="default"
                        defaultExpanded={false}
                        header="Who am I?"
                >
                        <ListGroup
                                bsClass="list-group"
                        >
                                <ListGroupItem
                                        bsClass="list-group-item"
                                        listItem={false}
                                >
                                        Name: 
                                        Fred
                                </ListGroupItem>
                                <ListGroupItem
                                        bsClass="list-group-item"
                                        listItem={false}
                                >
                                        Email: 
                                        fred_murtz@lucy.com
                                </ListGroupItem>
                        </ListGroup>
                </Panel>
        </Col>
</Row>,
      "componentClass": "div",
      "fluid": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Col
          bsClass="col"
          componentClass="div"
          sm={12}
>
          <Panel
                    bsClass="panel"
                    bsStyle="default"
                    defaultExpanded={false}
                    header="Who am I?"
          >
                    <ListGroup
                              bsClass="list-group"
                    >
                              <ListGroupItem
                                        bsClass="list-group-item"
                                        listItem={false}
                              >
                                        Name: 
                                        Fred
                              </ListGroupItem>
                              <ListGroupItem
                                        bsClass="list-group-item"
                                        listItem={false}
                              >
                                        Email: 
                                        fred_murtz@lucy.com
                              </ListGroupItem>
                    </ListGroup>
          </Panel>
</Col>,
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": <Panel
            bsClass="panel"
            bsStyle="default"
            defaultExpanded={false}
            header="Who am I?"
>
            <ListGroup
                        bsClass="list-group"
            >
                        <ListGroupItem
                                    bsClass="list-group-item"
                                    listItem={false}
                        >
                                    Name: 
                                    Fred
                        </ListGroupItem>
                        <ListGroupItem
                                    bsClass="list-group-item"
                                    listItem={false}
                        >
                                    Email: 
                                    fred_murtz@lucy.com
                        </ListGroupItem>
            </ListGroup>
</Panel>,
          "componentClass": "div",
          "sm": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "bsStyle": "default",
            "children": <ListGroup
              bsClass="list-group"
>
              <ListGroupItem
                            bsClass="list-group-item"
                            listItem={false}
              >
                            Name: 
                            Fred
              </ListGroupItem>
              <ListGroupItem
                            bsClass="list-group-item"
                            listItem={false}
              >
                            Email: 
                            fred_murtz@lucy.com
              </ListGroupItem>
</ListGroup>,
            "defaultExpanded": false,
            "header": "Who am I?",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "list-group",
              "children": Array [
                <ListGroupItem
                  bsClass="list-group-item"
                  listItem={false}
>
                  Name: 
                  Fred
</ListGroupItem>,
                <ListGroupItem
                  bsClass="list-group-item"
                  listItem={false}
>
                  Email: 
                  fred_murtz@lucy.com
</ListGroupItem>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "list-group-item",
                  "children": Array [
                    "Name: ",
                    "Fred",
                  ],
                  "listItem": false,
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  "Fred",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "list-group-item",
                  "children": Array [
                    "Email: ",
                    "fred_murtz@lucy.com",
                  ],
                  "listItem": false,
                },
                "ref": null,
                "rendered": Array [
                  "Email: ",
                  "fred_murtz@lucy.com",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": <Row
          bsClass="row"
          componentClass="div"
>
          <Col
                    bsClass="col"
                    componentClass="div"
                    sm={12}
          >
                    <Panel
                              bsClass="panel"
                              bsStyle="default"
                              defaultExpanded={false}
                              header="Who am I?"
                    >
                              <ListGroup
                                        bsClass="list-group"
                              >
                                        <ListGroupItem
                                                  bsClass="list-group-item"
                                                  listItem={false}
                                        >
                                                  Name: 
                                                  Fred
                                        </ListGroupItem>
                                        <ListGroupItem
                                                  bsClass="list-group-item"
                                                  listItem={false}
                                        >
                                                  Email: 
                                                  fred_murtz@lucy.com
                                        </ListGroupItem>
                              </ListGroup>
                    </Panel>
          </Col>
</Row>,
        "componentClass": "div",
        "fluid": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Col
            bsClass="col"
            componentClass="div"
            sm={12}
>
            <Panel
                        bsClass="panel"
                        bsStyle="default"
                        defaultExpanded={false}
                        header="Who am I?"
            >
                        <ListGroup
                                    bsClass="list-group"
                        >
                                    <ListGroupItem
                                                bsClass="list-group-item"
                                                listItem={false}
                                    >
                                                Name: 
                                                Fred
                                    </ListGroupItem>
                                    <ListGroupItem
                                                bsClass="list-group-item"
                                                listItem={false}
                                    >
                                                Email: 
                                                fred_murtz@lucy.com
                                    </ListGroupItem>
                        </ListGroup>
            </Panel>
</Col>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": <Panel
              bsClass="panel"
              bsStyle="default"
              defaultExpanded={false}
              header="Who am I?"
>
              <ListGroup
                            bsClass="list-group"
              >
                            <ListGroupItem
                                          bsClass="list-group-item"
                                          listItem={false}
                            >
                                          Name: 
                                          Fred
                            </ListGroupItem>
                            <ListGroupItem
                                          bsClass="list-group-item"
                                          listItem={false}
                            >
                                          Email: 
                                          fred_murtz@lucy.com
                            </ListGroupItem>
              </ListGroup>
</Panel>,
            "componentClass": "div",
            "sm": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "bsStyle": "default",
              "children": <ListGroup
                bsClass="list-group"
>
                <ListGroupItem
                                bsClass="list-group-item"
                                listItem={false}
                >
                                Name: 
                                Fred
                </ListGroupItem>
                <ListGroupItem
                                bsClass="list-group-item"
                                listItem={false}
                >
                                Email: 
                                fred_murtz@lucy.com
                </ListGroupItem>
</ListGroup>,
              "defaultExpanded": false,
              "header": "Who am I?",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "list-group",
                "children": Array [
                  <ListGroupItem
                    bsClass="list-group-item"
                    listItem={false}
>
                    Name: 
                    Fred
</ListGroupItem>,
                  <ListGroupItem
                    bsClass="list-group-item"
                    listItem={false}
>
                    Email: 
                    fred_murtz@lucy.com
</ListGroupItem>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "list-group-item",
                    "children": Array [
                      "Name: ",
                      "Fred",
                    ],
                    "listItem": false,
                  },
                  "ref": null,
                  "rendered": Array [
                    "Name: ",
                    "Fred",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "list-group-item",
                    "children": Array [
                      "Email: ",
                      "fred_murtz@lucy.com",
                    ],
                    "listItem": false,
                  },
                  "ref": null,
                  "rendered": Array [
                    "Email: ",
                    "fred_murtz@lucy.com",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
